python/cudf/cudf/core/series.py

```diff
@@ -3615,7 +3615,10 @@ def microsecond(self):
         """
         return Series(
             data=(
-                self.series._column.get_dt_field("millisecond")
+                # Need to manually promote column to int32 because
+                # pandas-matching binop behaviour requires that this
+                # __mul__ returns an int16 column.
+                self.series._column.get_dt_field("millisecond").astype("int32")
```

Comment:
This change is needed because previously this multiplication automatically promoted to int32, but now it doesn't (better matching pandas). If we don't like this, I think we need to admit defeat here and not match pandas corner-case for corner-case.

Reply:
FWIW, on pandas main, this returns int64, but that's because we unconditionally make these attributes return int64

https://github.com/pandas-dev/pandas/blob/f81f68720ed9d022c7f2416682d5485e609df9b6/pandas/_libs/tslibs/fields.pyx#L334

```
In [4]: datetime_series.dt.microsecond
Out[4]:
0    0
1    1
2    2
dtype: int64
```

Reply:
I think it makes sense to differ from pandas here because the consequence of using wider than necessary integer types is more problematic on GPU.